src/powershell-native/nativemsh/pwrshplugin/pwrshclrhost.cpp

```diff
@@ -37,6 +37,8 @@ unsigned int PowerShellCoreClrWorker::LaunchClr(
     _In_ LPCWSTR wszRuntimeVersion,
     _In_ LPCSTR friendlyName)
 {
+    // Allocate a console so that the codepage is setup correctly
+    AllocConsole();
```

Comment:
I wonder if there are any side effects with allocating a console in a non-interactive PowerShell remoting Windows session?  Probably not, but might be worth checking.

Reply:
Jordan did some ad hoc testing and verified it fixes the original issue and didn't see any side effects.  Allocating a console when using WinRM remoting actually provides consistency with local remoting across powershell processes which would have a console allocated implicitly.

Reply:
Should we be calling FreeConsole as well, maybe in the destructor?

Reply:
https://docs.microsoft.com/en-us/windows/console/freeconsole

Reply:
As noted in that documentation, the console is automatically detached when the last process in that tree terminates.

Reply:
I think Steve means _deleted_ not _detached_, from the docs:
> A console is closed when the last process attached to it terminates or calls FreeConsole.

To me the documentation implies `FreeConsole` is only used to detach from one console in order to create or attach to another console.

Reply:
I guess getting the code page set correctly is an undocumented side effect of `AllocConsole`? I mean, it makes sense that it does it, but the docs don't say anything about it. Still, considering the tests shows this works, I think it's the right solution. It's certainly unlikely to hurt anything.